R/unaids-navigator.R

```diff
@@ -53,16 +54,147 @@ write_navigator_checklist <- function(naomi_output,
              t_("NAVIGATOR_CAL_ART_DESC"),
              t_("NAVIGATOR_CAL_KOS_DESC"),
              t_("NAVIGATOR_CAL_NEW_INFECTIONS_DESC"),
-             t_("NAVIGATOR_CAL_METHOD_DESC"))                        
+             t_("NAVIGATOR_CAL_METHOD_DESC"))
 
   v <- data.frame(NaomiCheckPermPrimKey = key,
                   NaomiCheckDes = label,
-                  TrueFalse = NA)
+                  TrueFalse = FALSE)
 
+  ## Do checks
   ## These checks are always 'TRUE'
   always_true_checks <- c("Package_created", "Package_has_all_data", "Model_fit")
   v$TrueFalse[v$NaomiCheckPermPrimKey %in% always_true_checks] <- TRUE
 
+  
+  ## Check for correct model options selection
+  valid_opt <- yaml::read_yaml(system.file("metadata/navigator_validation.yml", package = "naomi"))
+
+  model_options <- naomi_output$fit$model_options
+  data_options <- naomi_output$fit$data_options
+  calibration_options <- naomi_output$fit$calibration_options
+
+  ## Compare aggregated naomi inputs to national Spectrum totals
+
+  v$TrueFalse[v$NaomiCheckPermPrimKey == "ART_is_Spectrum"] <- data_options$art_number_spectrum_aligned
+  v$TrueFalse[v$NaomiCheckPermPrimKey == "ANC_is_Spectrum"] <- data_options$anc_testing_spectrum_aligned
+
+
+  ## Is most recent calendar quarter selected
+  if (naomi_output$fit$model_options$calendar_quarter_t2 == valid_opt$calendar_quarter_t2) {
```

Comment:
I think these are better as `v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_recent_qtr"] <- naomi_output$fit$model_options$calendar_quarter_t2 == valid_opt$calendar_quarter_t2` (and that means code cov won't flag these lines as missing)

Reply:
addressed by 4143160